pkg/controller/storage/container/container.go

```diff
@@ -160,6 +161,10 @@ func (m *containerSyncdeleterMaker) newSyncdeleter(ctx context.Context, c *v1alp
 		return nil, errors.Wrapf(err, "failed to retrieve storage account reference: %s", c.Spec.AccountReference.Name)
 	}
 
+	if acct.GetWriteConnectionSecretToReference() == nil {
+		return nil, errors.New(errAcctSecretNil)
+	}
+
```

Comment:
Note: this is actually using the `Account`'s `writeConnectionSecretToRef`, not the `Provider` `credentialsSecretRef`. I noticed that we don't check for `nil` here though so i went ahead and updated.

Reply:
😬